plugins/govppmux/plugin_impl_govppmux.go

```diff
@@ -94,33 +100,30 @@ func (p *Plugin) Init() (err error) {
 	// register REST API handlers
 	p.registerHandlers(p.HTTPHandlers)
 
-	if p.vppAdapter == nil {
-		var address string
-		useShm := disabledSocketClient || p.config.ConnectViaShm || p.config.ShmPrefix != ""
-		if useShm {
-			address = p.config.ShmPrefix
-		} else {
-			address = p.config.BinAPISocketPath
-		}
-		p.vppAdapter = NewVppAdapter(address, useShm)
+	var address string
+	useShm := disabledSocketClient || p.config.ConnectViaShm || p.config.ShmPrefix != ""
```

Comment:
is it not possible that `disabledSocketClient` or `p.config.ConnectViaShm` will be true  and `p.config.ShmPrefix` will be an empty string?

Reply:
Sure it can and it will use default shmPrefix in that case.

Reply:
Here it is actually says it is okay not to set `ShmPrefix` and only set `ConnectViaShm` for some cases:
https://github.com/ligato/vpp-agent/blob/7a8451377a7ac27af23d1561e111f69d426969c2/plugins/govppmux/config.go#L30-L32
But I not follow if it is must be somewhere set to `/dev/shm` explicitly.

Reply:
IMHO, this abomination is here for some legacy reasons when we used `vppapiclient` (CGO wrapper on top of C library from VPP) to access VPP API. It probably isn't even relevant anymore and should be cleanup up.